### DISCUSSION TITLE
Merge grains having boundaries in common

### DISCUSSION THREAD

**Discussion by Saoussen-ouhiba** [mtex-toolbox/mtex#605] [Ask Anything] [answered]
Hello,

I have the following EBSD map.
![ebsd](https://user-images.githubusercontent.com/73701277/100638790-b735bb80-3334-11eb-9cd1-a03e9c156e9b.jpg)
I was able to determine island grains from this EBSD map as see in the following figure
![island grains 2](https://user-images.githubusercontent.com/73701277/100638904-daf90180-3334-11eb-90a3-6281be2f706d.jpg)
I would like now to merge the island grains that have boundaries in common. Is there a function to identify these grains please?
Thank you a lot,

Saoussen

**Comment by kilir** [comment, marked as answer]
Hi,
 if I understand you correctly, the following should be an example of what your try to do:
```
mtexdata twins;
grains = ebsd.calcGrains('angle',1*degree');
cinc = grains.isInclusion;

% that looks a bit complicated
gbm = grains(cinc).boundary(all(ismember(grains(cinc).boundary.grainId,grains(cinc).id),2))

migrains = merge(grains,gbm)

plot(grains(cinc))
hold on
plot(gbm,'linecolor','r')
hold on% merge inclusions
plot(migrains(migrains.isInclusion).boundary,'linecolor','g')
hold off
```
Cheers,
Rüdiger

**Reply by Saoussen-ouhiba** [reply]
Hello Rüdiger,

Thank you a lot for your help. It works great for me now.

Regards,

Saoussen
